### PR TITLE
Corrupt OMG handling

### PR DESCRIPTION
**Pull request by mountainraptor**
* Check for header before processing OMG
* Send errors from server to client in response
* Notify user of why OMG failed to load

### CODE REVIEW

**Comment by mountainraptor** [comment]
Merged new commits to avoid conflicts with the base branch.

File: lib/omgs/omgs-manager.js [hunk 1]
```diff
@@ -318,54 +324,55 @@ limitations under the License.
...
-
-        this._addOmg(omg)
-        .catch((err) => {
-          const filepath = omg.internal.omgFilepath;
-          logger.error('Failed to add %s: %s', filepath, err.message, {
-            error: {
-              name: err.name,
-              message: err.message,
-              stack: err.stack
+      return new Promise((resolve, reject) => {
```
Comment (KevinDyer) [inline lib/omgs/omgs-manager.js]: The design decision to not block the add OMG operation was made to due to a couple of concerns. Namely, OMGs have the potential to be large in size and disk I/O bound devices may not be able to complete the entire "add" OMG operation quickly. Due to this constraint, an omgInfo object containing state is return relatively quickly. A client can use the  omgInfo object to track state of the add operation. In the case of the upload request, perhaps we should change the client to receive the omgInfo object then notify the uploading user of and potential error that occurred.

**Comment by bbrother** [comment]
Sorry for the trouble we will not be accepting any pull requests that do not follow our conventions documented in the link below sorry for the trouble.  Please read this carefully before updating your pull request.

https://github.com/LGSInnovations/bits/wiki/Git-Commit-Message-Conventions

**Comment by nanpx** [comment]
@KevinDyer how do you want to proceed?

**Comment by nbroeking** [comment]
Closing because of age. Please address all issues and re-open pull request.

**Pull request closed without merging**